### PR TITLE
8 create UI tests and add to ci pipeline

### LINKED ISSUE #8: Create BDD tests and add to CI pipeline
Create BDD tests with Cucumber and add to CI pipeline

1.  Create BDD test for the application
2. When the test are created, modify the CI pipeline to run the tests automatically
3. Lastly ask @sutheo17 to review the changes

### PR DESCRIPTION
**Pull request by gerencsermate**
I added cucumber to the project.

Additionally implemented a few BDD test using the Cucumber java framework.

### CODE REVIEW

**Comment by gerencsermate** [comment]
I tested some more complex game scenarios with Cucumber.

The focus was on testing the movement of the saboteur and the mechanic when they meet on the pipe system.
Furthermore I tested pipe detaching.

**Comment by gerencsermate** [comment]
I wrote a short summary of the testing.

File: Drumakomi_sivatag/pom.xml
Comment (sutheo17) [inline Drumakomi_sivatag/pom.xml:1]: Thanks for adding Test Coverage to the pipeline.

**Review by sutheo17 — APPROVED** [review]
The tests seem to be running fine on my end. And the pipeline recognise them well fine.

**Pull request merged**